File: metamorph/samples/java/Matrix.java
public class Matrix {

    public static void main(String[] args) throws Exception {

        int[][] mat = new int[][] {
            {1, 2},
            {3, 4}
        };

        // Call the correct method
        String info = Probe.collectInfo(mat);

        System.out.println("[JAVA PROBE OUTPUT]");
        System.out.println(info);
    }
}

File: metamorph/samples/cpp/matrix.cpp
#include <iostream>
#include "../../probes/cpp_probe/probe.cpp"

int main() {
    int mat[2][2] = { {1, 2}, {3, 4} };

    std::string info = collectInfo(mat, typeid(mat).name());

    std::cout << "[C++ PROBE OUTPUT]\n";
    std::cout << info << std::endl;

    return 0;
}
